lib/src/cli/cli.dart

```diff
@@ -140,9 +140,10 @@ const _ignoredDirectories = {
   '.fvm',
 };
 
-bool _isPubspec(FileSystemEntity entity) {
+bool _isPubspec(FileSystemEntity entity, Set<String> ignore) {
   final segments = p.split(entity.path).toSet();
   if (segments.intersection(_ignoredDirectories).isNotEmpty) return false;
+  if (segments.intersection(ignore).isNotEmpty) return false;
```

Comment:
I don't think this will work if we pass a glob which I believe should be supported since most developers will likely expect globs to be supported.

```
flutter packages get -r --ignore "bricks/**"
```

Reply:
What do you think about this?
```dart
extension on Set<String> {
  bool excludes(FileSystemEntity entity) {
    final segments = p.split(entity.path).toSet();
    if (segments.intersection(_ignoredDirectories).isNotEmpty) return true;
    if (segments.intersection(this).isNotEmpty) return true;

    for (final value in this) {
      if (Glob(value).matches(entity.path)) {
        return true;
      }
    }

    return false;
  }
}
```

Reply:
Potentially could split this up if you think it's better 👍🏻 
```dart
for (final value in this) {
  final glob = Glob(value);
  if (glob.matches(entity.path)) {
    return true;
  }
}
```

Reply:
Yeah that seems reasonable to me